sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/ResourceType.cs

```diff
@@ -226,15 +226,15 @@ private void Parse(string resourceIdOrType)
                 throw new ArgumentOutOfRangeException(nameof(resourceIdOrType));
 
             // if the type is just subscriptions, it is a built-in type in the Microsoft.Resources namespace
-            if (parts.Count == 1)
+            else if (parts.Count == 1)
```

Comment:
I don't think this else is needed since the first if throws

Reply:
we need to make it "else if" otherwise the exception gets thrown from line 294

Reply:
For the if on 237 that might make sense, but it doesn't here.